kiva/fonttools/font_manager.py

```diff
@@ -47,1000 +29,58 @@
-                size = float(size)
-            except ValueError:
-                if size is not None and size not in font_scalings:
-                    raise ValueError("size is invalid")
-        self._size = size
-
-    def set_file(self, file):
-        """
-        Set the filename of the fontfile to use.  In this case, all
-        other properties will be ignored.
-        """
-        self._file = file
-
-    def copy(self):
-        """Return a deep copy of self"""
-        return FontProperties(_init=self)
-
-
-def ttfdict_to_fnames(d):
-    """
-    flatten a ttfdict to all the filenames it contains
-    """
-    fnames = []
-    for named in d.values():
-        for styled in named.values():
-            for variantd in styled.values():
-                for weightd in variantd.values():
-                    for stretchd in weightd.values():
-                        for fname in stretchd.values():
-                            fnames.append(fname)
-    return fnames
-
-
-def pickle_dump(data, filename):
-    """
-    Equivalent to pickle.dump(data, open(filename, 'wb'))
-    but closes the file to prevent filehandle leakage.
-    """
-    fh = open(filename, "wb")
-    try:
-        pickle.dump(data, fh)
-    finally:
-        fh.close()
-
-
-def pickle_load(filename):
-    """
-    Equivalent to pickle.load(open(filename, 'rb'))
-    but closes the file to prevent filehandle leakage.
+    font_manager : FontManager
     """
-    fh = open(filename, "rb")
-    try:
-        data = pickle.load(fh)
-    finally:
-        fh.close()
-    return data
+    global fontManager
+    if fontManager is None:
+        fontManager = _load_from_cache_or_rebuild(_get_font_cache_path())
+    return fontManager
 
 
 class FontManager:
+    """ The :class:`FontManager` singleton instance is created with a list of
+    TrueType fonts based on the font properties: name, style, variant, weight,
+    stretch, and size. The :meth:`findfont` method does a nearest neighbor
+    search to find the font that most closely matches the specification. If no
+    good enough match is found, a default font is returned.
     """
-    On import, the :class:`FontManager` singleton instance creates a
-    list of TrueType fonts based on the font properties: name, style,
-    variant, weight, stretch, and size.  The :meth:`findfont` method
-    does a nearest neighbor search to find the font that most closely
-    matches the specification.  If no good enough match is found, a
-    default font is returned.
-    """
-
     # Increment this version number whenever the font cache data
     # format or behavior has changed and requires a existing font
     # cache files to be rebuilt.
-    __version__ = 8
+    __version__ = 9
 
     def __init__(self, size=None, weight="normal"):
         self._version = self.__version__
 
         self.__default_weight = weight
-        self.default_size = size
+        self.default_size = size if size is not None else 12.0
```

Comment:
why not set `size=12.0` in the method signature?
Also why size and default_size? it seems in the rest of the code for the class default_size is used

Reply:
I'm glad you brought this up because I forgot about it. It's because of this code: https://github.com/enthought/enable/pull/704/files#diff-07c82150ae0bd465451ecaaaf02e959e718f46ea5e494d84dfd2ed81e6194316L1216-L1217.

That line of code in `score_size` gets run if you pass a a `FontProperties` instance to `FontManager.findfont` with its `size` attribute explicitly set to one of the strings in `font_scalings`. That's basically never. I know, because nobody complained about it in the 11 years that it had a bug that would raise an exception.

I fixed that bug, but now `score_size` must be called with a valid default size. That means that `FontManager` needs to have a non-None value set to `.default_size`. **But it gets worse.** `_load_from_cache_or_rebuild` _explicitly_ sets `FontManager.default_size` back to None after loading it from the cache! We should probably stop doing that, but the fact that it's done implies that messing with the `default_size` was somehow the expected interface?

I'm not sure what to do.

Reply:
I see you've opened #709, thank you!  This can be addressed separately from this PR

Reply:
#709 opened to track